### PR TITLE
Notify option to determine when code is live

### PR DESCRIPTION
**Pull request by travisofthenorth**
This PR adds an additional notify option (`-n`) to dual burner deploys. By specifying this option, as well as a "version" endpoint for the apps in the cluster which returns the currently SHA, this PR will curl the specified endpoints until they match the deployed SHA.

@HoyaBoya @prsimp

### CODE REVIEW

**Comment by travisofthenorth** [comment]
Specs on the way

File: lib/dual_burner/commands/deployer.rb [hunk 1]
```diff
@@ -23,15 +28,50 @@ def deploy(git_reference: 'master')
...
+      def wait_for_deployment(git_reference)
+        deployed_sha = `git log #{git_reference} -n 1 --pretty=format:'%H'`
+        start_time = Time.now
+
+        while current_app_versions.any? { |sha| sha != deployed_sha }
+          sleep 2
+
+          # Give up after 5 minutes
+          if Time.now - start_time > 300
+            puts "Your code still isn't live...I give up."
```
Comment (midu) [inline lib/dual_burner/commands/deployer.rb]: I thought it could take up to 10min...

Comment (HoyaBoya) [inline lib/dual_burner/commands/deployer.rb]: Do we want to Terminal notify a "womp womp" if we take this long? We not just loop infinitely and make people CTL-C out of things?

I guess I see this as pork-n-porky to maybe work and maybe not work for a successful deploy. Non-determinism feels sorta bad for a notification.

File: lib/dual_burner/commands/deployer.rb [hunk 2]
```diff
@@ -23,15 +28,50 @@ def deploy(git_reference: 'master')
...
+        TerminalNotifier.notify(
+          "Latest version (#{git_reference}) is now live!",
+          title: "#{env} deploy",
+          sound: 'alarm'
+        )
+      end
+
+      def current_app_versions
+        results = execute_command_with_threads do |app|
+          "curl -s #{app.version_url} -w \"\n\""
```
Comment (HoyaBoya) [inline lib/dual_burner/commands/deployer.rb]: This was the issue? Ba humbug.

**Comment by HoyaBoya** [comment]
Nice work here dude. I am pretty dependent on my `prod_version` and `sage_version` script to know when to test things, verify, etc. 👍,looks like you are paying for some 👮 debt from the past.

**Comment by travisofthenorth** [comment]
@HoyaBoya added a spec

**Comment by HoyaBoya** [comment]
Seems good to me. Give it a whirl with a deploy to travis-core and push it!

**Pull request merged**